src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/RepresentativeService.java

```diff
@@ -38,11 +44,23 @@ public Representative getApplicantRepresentative(CaseData caseData) {
         var organisationId = caseData.getApplicant1OrganisationPolicy().getOrganisation().getOrganisationID();
         var representative = fromOrganisation(organisationService.findOrganisationById(organisationId)
                                                   .orElseThrow(RuntimeException::new));
+
+        var providedSolicitor1ServiceAddress = caseData.getApplicantSolicitor1ServiceAddress();
+
         return representative.toBuilder()
             .emailAddress(caseData.getApplicantSolicitor1UserDetails().getEmail())
+            .serviceAddress(fromProvidedAddress(representative.getServiceAddress(), providedSolicitor1ServiceAddress))
             .build();
     }
 
+    private Address fromProvidedAddress(
+        Address registeredServiceAddress, SolicitorServiceAddress providedServiceAddress) {
+        if (providedServiceAddress != null && providedServiceAddress.getHasServiceAddress().equals(YesOrNo.YES)) {
+            return providedServiceAddress.getAddress();
+        }
+        return registeredServiceAddress;
```

Comment:
We could make it a bit more readable:
```
return Optional.ofNullable(providedServiceAddress)
    .map(SolicitorServiceAddress::getAddress)
    .orElse(registeredServiceAddress);
```
`hasServiceAddress` check is redundant if it's used as show condition for address field

Reply:
providedServiceAddress null check is just backward compatibility for releasing service code first, otherwise it will be mandatory field